src/components/Whiteboard/ElementBehaviors/Selection/calculateIntersect.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Element, calculateBoundingRectForElements } from '../../../../state';
+import { Elements } from '../../../../state/types';
+
+/**
+ * Calculates which elements overlap a selection using their bounding rectangles.
+ *
+ * @param selection - selection element
+ * @param elements - elements to test for intersection
+ * @returns IDs of the intersecting elements
+ */
+export function calculateIntersect(
+  selection: Element,
+  elements: Elements,
+): string[] {
+  const selectedElementIds = [];
```

Comment:
I think we should be consistent with the codebase and be specific about the string array type here.